Pet Mod Support (Alternatium [PMOTS])

yakatk1012 opened the pull request:
Hello,

I tried my best coding teambuilder support for Alternatium, and to make a template for future PMOTS/Leader's Choice Pet Mods to follow. There's only two things that are missing right now:
1. The custom Pokemon do not show up in the formats. You have to type them in to make them show up, and they do not show the correct tier.
2. The formats repeat themselves multiple times. I do not know what is causing this.

Let me know if you have any suggestions!

yakatk1012 committed:
BattleTeambuilderTable support for Pet Mods
Add initial teambuilder support for PMOTS/Leader's Choice (not showing, yet)

yakatk1012 committed:
Add teambuilder support for Pet Mods (Alternatium [PMOTS])
Add everything except for support for custom Pokemon for Alternatium

KrisXV commented on build-tools/build-indexes:
```diff
@@ -321,12 +346,14 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 		const isLetsGo = (genIdent === 7.1);
 		const isBDSP = (genIdent === 8.6 || genIdent === -8.6);
+		const isPMOTS = (genIdent === 8.7);
+		const isLC = (genIdent === 8.8);
```
LC means little cup in most context, use isPetModLC here please

KrisXV commented on build-tools/build-indexes:
```diff
@@ -516,6 +545,22 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 			BattleTeambuilderTable['gen8bdsp'].overrideTier = overrideTier;
 			BattleTeambuilderTable['gen8bdsp'].monotypeBans = monotypeBans;
 			BattleTeambuilderTable['gen8bdsp'].formatSlices = formatSlices;
+		} else if (isPMOTS) {
+			BattleTeambuilderTable['alternatium'] = {};
+			BattleTeambuilderTable['alternatium'].learnsets = {};
+			BattleTeambuilderTable['alternatium'].tiers = tiers;
+			BattleTeambuilderTable['alternatium'].items = items;
+			BattleTeambuilderTable['alternatium'].overrideTier = overrideTier;
+			BattleTeambuilderTable['alternatium'].monotypeBans = monotypeBans;
+			BattleTeambuilderTable['alternatium'].formatSlices = formatSlices;
+		} else if (isLC) {
+			BattleTeambuilderTable['randomdex'] = {};
+			BattleTeambuilderTable['randomdex'].learnsets = {};
+			BattleTeambuilderTable['randomdex'].tiers = tiers;
+			BattleTeambuilderTable['randomdex'].items = items;
+			BattleTeambuilderTable['randomdex'].overrideTier = overrideTier;
+			BattleTeambuilderTable['randomdex'].monotypeBans = monotypeBans;
+			BattleTeambuilderTable['randomdex'].formatSlices = formatSlices;
```
why not make these tables just say 'pmots' and 'petmodlc'

KrisXV commented on build-tools/build-indexes:
```diff
@@ -555,6 +600,9 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 			if (isLetsGo) {
 				return ["Uber", "OU", "UU", "NFE", "LC"];
 			}
+			if (isPMOTS) {
+				return ["OU"];
+			}
```
this isnt necessary, it uses the default gen 8 table (which includes OU)

Petuuuhhh replied:
In Alternatium's case, it only contains OU mons, so this is necessary.

KrisXV replied:
No it’s not, look at how nfe does it

KrisXV commented on build-tools/build-indexes:
```diff
@@ -906,14 +954,22 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 			BattleTeambuilderTable['gen7letsgo'].learnsets[id][moveid] = '7';
 		}
 	}
-	const BDSPLearnsets = Dex.mod('gen8bdsp').data.Learnsets;
-	for (const id in BDSPLearnsets) {
-		const species = Dex.mod('gen8bdsp').species.get(id);
-		if (species.isNonstandard && !['Unobtainable', 'CAP'].includes(species.isNonstandard)) continue;
-		const learnset = BDSPLearnsets[id].learnset;
-		BattleTeambuilderTable['gen8bdsp'].learnsets[id] = {};
+	const PMOTSLearnsets = Dex.mod('alternatium').data.Learnsets;
+	for (const id in PMOTSLearnsets) {
+		const species = Dex.mod('alternatium').species.get(id);
+		const learnset = PMOTSLearnsets[id].learnset;
+		BattleTeambuilderTable['alternatium'].learnsets[id] = {};
+		for (const moveid in learnset) {
+			BattleTeambuilderTable['alternatium'].learnsets[id][moveid] = '8';
+		}
+	}
+	const LCLearnsets = Dex.mod('randomdex').data.Learnsets;
+	for (const id in LCLearnsets) {
+		const species = Dex.mod('randomdex').species.get(id);
+		const learnset = LCLearnsets[id].learnset;
+		BattleTeambuilderTable['randomdex'].learnsets[id] = {};
 		for (const moveid in learnset) {
-			BattleTeambuilderTable['gen8bdsp'].learnsets[id][moveid] = '8g';
+			BattleTeambuilderTable['randomdex'].learnsets[id][moveid] = '8';
```
._. why are you removing bdsp support

Petuuuhhh replied:
my bad, I overlooked that

KrisXV commented on build-tools/build-indexes:
```diff
@@ -1046,7 +1102,7 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 	//
 
-	for (const mod of ['gen7letsgo', 'gen8bdsp']) {
+	for (const mod of ['gen7letsgo', 'gen8bdsp', 'alternatium', 'randomdex']) {
```
this is more server-side, but we should probably make these mod files pmots and pmlc to make support consistent in future cases

KrisXV commented on build-tools/build-indexes:
```diff
@@ -1057,13 +1113,41 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 		for (const id in modData.Pokedex) {
 			const modEntry = modData.Pokedex[id];
 			const parentEntry = parentData.Pokedex[id];
+			if (typeof parentEntry === 'undefined') {
```
```suggestion
			if (!parentEntry) {
```

yakatk1012 committed:
Review edits.
Added back BDSP support for learnsets (sorry!) and changed things up to be more in line with general PMOTS/Leader's Choice support. Also changed isLC to isPetModLC for clarification.

Petuuuhhh commented:
I'd like to point out that the custom Pokemon added by Alternatium show up if you search for them in other formats.

BDSP:
![deoxys_customs_bdsp](https://user-images.githubusercontent.com/90161287/152686790-99267b42-4b6c-4b23-80dc-c70ef173205f.png)

National Dex:
![deoxys_customs_natdex](https://user-images.githubusercontent.com/90161287/152686799-def728a3-7a3a-42c1-b47d-f3b83e1cf4c8.png)

I am not sure how to disable this.

yakatk1012 committed:
Follow Kris's advice on PMOTS tiers
Make PMOTS only show OU mons, while making all other mons illegal.

AnnikaCodes commented:
Alternatium isn't a pet mod any more so I am going to mark this as a draft.
